dev/tools/gen_defaults/lib/fab_template.dart

```diff
@@ -47,6 +46,7 @@ class _TokenDefaultsM3 extends FloatingActionButtonThemeData {
 
   bool get _isExtended => type == _FloatingActionButtonType.extended;
 
+  @override double? get elevation => Scaffold.of(context).fabIsContained ? 0.0 : ${elevation("md.comp.fab.primary.container")};
```

Comment:
I am still thinking this is coupling the Scaffold too closely to the FAB (I know it is already coupled, but this would just be adding to it).  Perhaps this can just be handed in the example case, but just turning off the elevation when we are positioning it in the bottom app bar.

Reply:
I agree with you, I was able to specify the location of FAB, but that also will be a coupling.